include/networkit/distance/AStar.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * AStar.hpp
+ *
+ *  Created on: 09.08.2019
+ *      Author: Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+#ifndef A_STAR_H
+#define A_STAR_H
+
+#include <networkit/distance/AStarGeneral.hpp>
+
+/**
+ * @ingroup distance
+ * AStar path-finding algoritm
+ */
+namespace NetworKit {
+class AStar final : public STSP {
+public:
+    /**
+     * Creates the AStarGeneral class for the graph @a G, the source node @a
+     * source, and the target node @a target using a template parameter as
+     * heuristic function.
+     *
+     * @param G The graph.
+     * @param distanceHeu Vector with a lower bound of the distance from every
+     * node to the target.
+     * @param source The source node.
+     * @param target The target node.
+     * @param storePred If true, the algorithm will also store the predecessors
+     * and reconstruct a shortest path from @a source and @a target.
+     */
+    AStar(const Graph &G, const std::vector<double> &distanceHeu, node source, node target,
+                 bool storePred = true)
+        : STSP(G, source, target, storePred), distanceHeu(distanceHeu),
+          astar(AStarGeneral<Heuristic>(G, Heuristic(distanceHeu), source, target, storePred)) {}
+
+    /*
+     * Executes the AStar algorithm.
+     */
+    void run() override {
+        astar.run();
+        hasRun = true;
+    }
+
+    edgeweight getDistance() const override { return astar.getDistance(); }
+
+    std::vector<node> getPath() const { return astar.getPath(); }
+
+private:
+    const std::vector<double> &distanceHeu;
+    struct Heuristic {
+    public:
+        Heuristic(const std::vector<double> &distanceHeu) : distanceHeu(distanceHeu) {}
+        bool operator()(node u) { return distanceHeu[u]; }
```

Comment:
Why does the Function return bool? Should be `double`!
Functor should be const (and noexcept)